vitest.config.ts

```diff
@@ -0,0 +1,14 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
+  test: {
+    environment: "happy-dom",
+    include: ["./src/tests/unit/*.test.(ts|tsx)"],
+    coverage: {
+      all: true,
+      provider: "istanbul",
+      reporter: ["clover"],
+      include: ["src/**/*.{ts,tsx}"]
```

Comment:
We probably want to exclude (some of) the code in `src/core` from code coverage.